workers/unity/Assets/Playground/Scripts/Cubes/CollisionProcessSystem.cs

```diff
@@ -0,0 +1,68 @@
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Playground
+{
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    internal class CollisionProcessSystem : ComponentSystem
+    {
+        private struct LaunchableData
+        {
+            public readonly int Length;
+            public ComponentDataArray<SpatialOSLaunchable> Launchable;
+            public ComponentDataArray<CollisionComponent> Collision;
+            [ReadOnly] public ComponentDataArray<CommandRequestSender<SpatialOSLauncher>> Sender;
+        }
+
+        [Inject] private LaunchableData launchableData;
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < launchableData.Length; i++)
+            {
+                var collision = launchableData.Collision[i];
+                var first = launchableData.Launchable[i];
+                var second = collision.other;
+                var first_launcher = first.MostRecentLauncher;
+                var second_launcher = second.MostRecentLauncher;
+                if (first_launcher == 0 && second_launcher == 0)
+                {
+                    continue;
+                }
+                if (first_launcher != 0 && second_launcher != 0 && first_launcher != second_launcher)
+                {
+                    first.MostRecentLauncher = 0;
+                    second.MostRecentLauncher = 0;
+                    launchableData.Launchable[i] = first;
+                    collision.other = second;
+                    launchableData.Collision[i] = collision;
+                    continue;
```

Comment:
Would it be possible to use else if instead of continue in these statements?

Reply:
I tend to prefer not using continue, break and similar, though this seemed fairly natural. Will change it.